observatory-platform/observatory/platform/cli/generate_command.py

```diff
@@ -73,95 +73,119 @@ def generate_terraform_config(self, config_path: str):
-        telescope_path, dag_path, test_path, doc_path = self.get_telescope_template_path_(telescope_type)
-        telescope_module = telescope_name.lower()
-        telescope_file = f"{telescope_module}.py"
-
-        # Render dag
-        dag = render_template(dag_path, telescope_module=telescope_module, telescope_name=telescope_name)
-
-        # Render telescope
-        telescope = render_template(telescope_path, telescope_name=telescope_name)
-
-        # Render test
-        test = render_template(test_path, telescope_name=telescope_name)
-
-        # Render documentation
-        doc = render_template(doc_path, telescope_name=telescope_name)
-
-        # Save templates
-        dag_dst_dir = observatory.dags.dags.__path__[0]
-        dag_dst_file = os.path.join(dag_dst_dir, telescope_file)
-
-        telescope_dst_dir = observatory.dags.telescopes.__path__[0]
-        telescope_dst_file = os.path.join(telescope_dst_dir, telescope_file)
-
-        test_dst_dir = "tests/observatory/dags/telescopes"
-        test_dst_file = os.path.join(test_dst_dir, f"test_{telescope_module}.py")
-
-        doc_dst_dir = "docs"
-        doc_dst_file = os.path.join(doc_dst_dir, "telescopes", f"{telescope_module}.md")
-
-        doc_index_file = os.path.join(doc_dst_dir, "telescopes", "index.rst")
-
-        print(f"Created a new dag file: {dag_dst_file}")
-        print(f"Created a new telescope file: {telescope_dst_file}")
-        print(f"Created a new telescope documentation file: {doc_dst_file}")
-        print(f"Created a new telescope test file: {test_dst_file}")
+        # Standardise names
+        package_name = stringcase.snakecase(package_name)
+        workflow_class = stringcase.capitalcase(stringcase.camelcase(workflow_name))
+        workflow_module = stringcase.lowercase(stringcase.snakecase(workflow_name))
+        workflow_file = f"{workflow_module}.py"
+
+        # Destination folders e.g.
+        # academic-observatory-workflows/academic_observatory_workflows/workflows
+        # academic-observatory-workflows/academic_observatory_workflows/dags
+        # academic-observatory-workflows/tests/academic_observatory_workflows/workflows
+        # academic-observatory-workflows/docs
+        workflow_dst_dir = os.path.join(project_path, package_name, "workflows")
+        dags_dst_dir = os.path.join(project_path, package_name, "dags")
+        test_dst_dir = os.path.join(project_path, "tests", package_name, "workflows")
+        doc_dst_dir = os.path.join(project_path, "docs")
+
+        # Make folders
+        for path in [workflow_dst_dir, dags_dst_dir, test_dst_dir, doc_dst_dir]:
+            os.makedirs(path, exist_ok=True)
+
+        # Make init files
+        package_folder = os.path.join(project_path, package_name)
+        tests_package_folder = os.path.join(project_path, "tests", package_name)
+        init_paths = [package_folder, tests_package_folder, workflow_dst_dir, dags_dst_dir, test_dst_dir]
+        for path in init_paths:
```

Comment:
create only if it doesn't exist

Reply:
This will be covered by the changes in the telescope documentation branch, at least that is what I'm assuming, because there are still quite a lot of difference in the `generate workflow` command for these two branches.